presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraSplitManager.java

```diff
@@ -45,23 +46,37 @@
     private final CassandraSession cassandraSession;
     private final int partitionSizeForBatchSelect;
     private final CassandraTokenSplitManager tokenSplitMgr;
+    private final CassandraPartitionManager partitionManager;
 
     @Inject
     public CassandraSplitManager(
             CassandraClientConfig cassandraClientConfig,
             CassandraSession cassandraSession,
-            CassandraTokenSplitManager tokenSplitMgr)
+            CassandraTokenSplitManager tokenSplitMgr,
+            CassandraPartitionManager partitionManager)
     {
         this.cassandraSession = requireNonNull(cassandraSession, "cassandraSession is null");
         this.partitionSizeForBatchSelect = cassandraClientConfig.getPartitionSizeForBatchSelect();
         this.tokenSplitMgr = tokenSplitMgr;
+        this.partitionManager = requireNonNull(partitionManager, "partitionManager is null");
     }
 
     @Override
     public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableHandle tableHandle, SplitSchedulingStrategy splitSchedulingStrategy)
     {
         CassandraTableHandle cassandraTableHandle = (CassandraTableHandle) tableHandle;
-        List<CassandraPartition> partitions = cassandraTableHandle.getPartitions();
+
+        List<CassandraPartition> partitions;
+        String clusteringKeyPredicates;
+        if (cassandraTableHandle.getPartitions().isPresent()) {
+            partitions = cassandraTableHandle.getPartitions().get();
+            clusteringKeyPredicates = cassandraTableHandle.getClusteringKeyPredicates();
```

Comment:
Looks like `clusteringKeyPredicates` should be Optional too.

Reply:
Let's keep it minimal